mappers/setvalue.go

```diff
@@ -11,13 +11,13 @@ type SetValue struct {
 }
 
 func (d SetValue) FromInternal(data map[string]interface{}) {
-	if d.ExternalValue != nil {
+	if len(data) != 0 && d.ExternalValue != nil {
```

Comment:
Can we do nil check? non-nil map can have zero len.

Reply:
fixed